test/spec/plugin_test.js

```diff
@@ -165,6 +165,30 @@ describe('webdriverajax', function testSuite() {
       assert.equal(request.response.headers['content-length'], contentLength);
     });
 
+    it('can access a certain angular 10 http client request', async function () {
+      await browser.url('/angular10.html');
+      await browser.setupInterceptor();
+      await completedRequest('#button');
+      const request = await browser.getRequest(0);
+      assert.equal(request.method, 'GET');
+      assert.equal(request.url, '/get.json');
+      assert.deepEqual(request.response.body, { OK: true });
+      assert.equal(request.response.statusCode, 200);
+      assert.equal(request.response.headers['content-length'], contentLength);
+    });
+
+    it('can access a certain angular 12 http client request', async function () {
```

Comment:
Since these tests are otherwise equivalent (and any changes to one should conceptually also be done for the other), we should parameterize them:

```js
describe('Angular compatibility', function () {
  [10, 12].forEach((version) => {
    it(`can assess XHR calls made in Angular ${version}`, function () {
      // Implementation
    });
    // A test for two simultaneous calls in flight?
  });
});
```

Reply:
Here you go https://github.com/webdriverio-community/wdio-intercept-service/pull/225/commits/49224cb242c88664b693fab019d9d7d3c8233b5f

Reply:
Thanks!

BTW the comment suggestion about two requests was more for assurances that the details are not mixed: two of the same exact request, waiting for each to fully complete before launching the next, cannot dispel any concurrency concerns.

(To do that you'd want to initiate a slow request and then a fast request, and require that the retrieved requests match up correctly.)

Reply:
Oh I see, will update that.

Reply:
I re-used the same logic as the tests that check ordering -> https://github.com/webdriverio-community/wdio-intercept-service/pull/225/commits/f07adb6457689538ee8167d6504f39bbe08ecbda
Let me know if it's ok.

Reply:
What's interesting to me is that the 2 POST requests pass without the change to the interceptor, while the GET request fails. Seems like maybe Angular 10 wasn't calling `xhr.abort()` with POST requests?

(As an aside, I wasn't able to get the wdio-static-server middleware to work for GET requests.)

Reply:
You're right, didn't notice that straight away! But it feels strange because in the angular fix they're using a POST request in the tests https://github.com/angular/angular/commit/a5d5f67be7407be0e6eb2c059356c23e83049b13